src/Microsoft.AspNet.OData.Shared/Query/Expressions/SelectExpandBinder.cs

```diff
@@ -349,7 +350,11 @@ internal Expression ProjectElement(Expression source, SelectExpandClause selectE
 
             Type wrapperGenericType = GetWrapperGenericType(isInstancePropertySet, isTypeNamePropertySet, isContainerPropertySet);
             wrapperType = wrapperGenericType.MakeGenericType(elementType);
-            return Expression.MemberInit(Expression.New(wrapperType), wrapperTypeMemberAssignments);
+
+            if (instanseOfWrapperClassWasSet)
+                return Expression.MemberInit(Expression.New(wrapperType), wrapperTypeMemberAssignments);
```

Comment:
as per code guidelines, please enclose body of conditional in braces (even if it's a single line).  Improves code readability and reduces introducing logical errors later.